packages/dds/matrix/src/test/time/sharedMatrix.spec.ts

```diff
@@ -40,242 +43,813 @@ describe("SharedMatrix execution time", () => {
+						minBatchDurationSeconds: 0,
+						maxBenchmarkDurationSeconds: matrixSize === 100 ? 10 : 5,
+					});
 				});
 
 				// Test the execute time of the SharedMatrix for removing a row in the middle for a given number of times.
-				benchmark({
-					type: BenchmarkType.Measurement,
-					title: `Remove the middle row ${count} times`,
-					benchmarkFnCustom: async <T>(state: BenchmarkTimer<T>) => {
-						let duration: number;
-
-						do {
-							// Since this setup one collects data from one iteration, assert that this is what is expected.
-							assert.equal(state.iterationsPerBatch, 1);
-
-							// Setup
-							localMatrix = createLocalMatrix({
-								id: "testLocalMatrix",
-								size: matrixSize,
-								initialValue: matrixValue,
-							});
-
-							// Operation
-							const before = state.timer.now();
-							for (let i = 0; i < count; i++) {
-								localMatrix.removeRows(Math.floor(localMatrix.rowCount / 2), 1);
-							}
-							const after = state.timer.now();
-
-							// Measure
-							duration = state.timer.toSeconds(before, after);
-
-							// Collect data
-						} while (state.recordBatch(duration));
-					},
-					// Force batch size of 1
-					minBatchDurationSeconds: 0,
-					// Matrix size 100 benchmarks use increased duration (10s) to improve statistical significance.
-					maxBenchmarkDurationSeconds: matrixSize === 100 ? 10 : 5,
+				describe(`Row Removal`, () => {
+					benchmark({
+						type: BenchmarkType.Measurement,
+						title: `Remove the middle row ${count} times`,
+						benchmarkFnCustom: async <T>(state: BenchmarkTimer<T>) => {
+							let duration: number;
+
+							do {
+								// Since this setup one collects data from one iteration, assert that this is what is expected.
+								assert.equal(state.iterationsPerBatch, 1);
+
+								// Setup
+								localMatrix = createLocalMatrix({
+									id: "testLocalMatrix",
+									size: matrixSize,
+									initialValue: matrixValue,
+								});
+
+								// Operation
+								const before = state.timer.now();
+								for (let i = 0; i < count; i++) {
+									localMatrix.removeRows(Math.floor(localMatrix.rowCount / 2), 1);
+								}
+								const after = state.timer.now();
+
+								// Measure
+								duration = state.timer.toSeconds(before, after);
+
+								// Collect data
+							} while (state.recordBatch(duration));
+						},
+						// Force batch size of 1
+						minBatchDurationSeconds: 0,
+						// Matrix size 100 benchmarks use increased duration (10s) to improve statistical significance.
+						maxBenchmarkDurationSeconds: matrixSize === 100 ? 10 : 5,
+					});
+
+					benchmark({
+						type: BenchmarkType.Measurement,
+						title: `Undo remove the middle row ${count} times`,
+						benchmarkFnCustom: async <T>(state: BenchmarkTimer<T>) => {
+							let duration: number;
+
+							do {
+								// Since this setup one collects data from one iteration, assert that this is what is expected.
```

Comment:
We repeat this pattern in most if not all of the tests:
```typescript
do {
								<SETUP>

								const before = state.timer.now();
								for (let i = 0; i < count; i++) {
									<OPERATION>
								}
								const after = state.timer.now();

								// Measure
								duration = state.timer.toSeconds(before, after);

								// Collect data
							} while (state.recordBatch(duration));
```

We should probably extract this into a helper function that all of the tests can call with their own setup / operation functionality.